compilation issue step-75 on ARM clang 13

```
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/allocator.h:154:28: error: no matching constructor for initialization of 'std::shared_ptr<const dealii::Triangulation<2, 2>>'

        ::new ((void*)__p) _Up(_VSTD::forward<_Args>(__args)...);

                           ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/allocator_traits.h:290:13: note: in instantiation of function template specialization 'std::allocator<std::shared_ptr<const dealii::Triangulation<2, 2>>>::construct<std::shared_ptr<const dealii::Triangulation<2, 2>>, dealii::Triangulation<2, 2> *, (lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9)>' requested here

        __a.construct(__p, _VSTD::forward<_Args>(__args)...);

            ^

/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/vector:948:21: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<std::shared_ptr<const dealii::Triangulation<2, 2>>>>::construct<std::shared_ptr<const dealii::Triangulation<2, 2>>, dealii::Triangulation<2, 2> *, (lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9), void>' requested here

    __alloc_traits::construct(this->__alloc(), _VSTD::__to_address(__tx.__pos_),

                    ^

/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/vector:1706:9: note: in instantiation of function template specialization 'std::vector<std::shared_ptr<const dealii::Triangulation<2, 2>>>::__construct_one_at_end<dealii::Triangulation<2, 2> *, (lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9)>' requested here

        __construct_one_at_end(_VSTD::forward<_Args>(__args)...);

        ^

/Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:671:34: note: in instantiation of function template specialization 'std::vector<std::shared_ptr<const dealii::Triangulation<2, 2>>>::emplace_back<dealii::Triangulation<2, 2> *, (lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9)>' requested here

      coarse_grid_triangulations.emplace_back(

                                 ^

/Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:1239:5: note: in instantiation of function template specialization 'Step75::solve_with_gmg<dealii::LinearAlgebra::distributed::Vector<double>, Step75::LaplaceOperator<2, double>, 2>' requested here

    solve_with_gmg(solver_control,

    ^

/Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:1473:9: note: in instantiation of member function 'Step75::LaplaceProblem<2>::solve_system' requested here

        solve_system();

        ^

/Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:1506:23: note: in instantiation of member function 'Step75::LaplaceProblem<2>::run' requested here

      laplace_problem.run();

                      ^

/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:469:26: note: candidate constructor template not viable: no known conversion from 'dealii::Triangulation<2, 2> *' to 'std::nullptr_t' (aka 'nullptr_t') for 1st argument

    template <class _Dp> shared_ptr(nullptr_t __p, _Dp __d);

                         ^

/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:464:9: note: candidate template ignored: requirement '__shared_ptr_deleter_ctor_reqs<(lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9), dealii::Triangulation<2, 2>, const dealii::Triangulation<2, 2>>::value' was not satisfied [with _Yp = dealii::Triangulation<2, 2>, _Dp = (lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9)]

        shared_ptr(_Yp* __p, _Dp __d,
```

problematic line:
https://github.com/dealii/dealii/blob/e6779913e7f770652da442b3c2460dec3e864029/examples/step-75/step-75.cc#L673

relevant error:
```
shared_ptr.h:464:9: note: candidate template ignored: requirement '__shared_ptr_deleter_ctor_reqs<(lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9), dealii::Triangulation<2, 2>, const dealii::Triangulation<2, 2>>::value' was not satisfied [with _Yp = dealii::Triangulation<2, 2>, _Dp = (lambda at /Users/heister/libs-candi-master-v5/tmp/unpack/deal.II-master/examples/step-75/step-75.cc:673:9)]

        shared_ptr(_Yp* __p, _Dp __d,
```

--- a/examples/step-75/step-75.cc
+++ b/examples/step-75/step-75.cc
@@ -669,8 +669,7 @@ namespace Step75
           dof_handler.get_triangulation());
     else
       coarse_grid_triangulations.emplace_back(
-        const_cast<Triangulation<dim> *>(&(dof_handler.get_triangulation())),
-        [](auto &) {});
+        &(dof_handler.get_triangulation()), [](auto *) {});
 
     // Determine the total number of levels for the multigrid operation and
     // allocate sufficient memory for all levels.

